docs/rendering_system.rst

```diff
@@ -0,0 +1,241 @@
+This transform validates that the required headers are present and of the
+correct data type, and removes headers not for display.
+It must run before the ``PEPTitle`` transform.
+
+
+3.4.3 ``PEPTitle`` transform
+****************************
+
+We generate the title node from the parsed title in the PEP headers, and make
+all nodes in the document children of the new title node.
+This transform must also handle parsing reStructuredText markup within PEP
+titles, such as :doc:`PEP 604 <../pep-0604>`.
+
+
+3.4.4 ``PEPContents`` transform
+*******************************
+
+The automatic table of contents (TOC) is inserted in this transform in a two
+part process.
+
+Firstly the transform inserts a placeholder for the TOC and a horizontal rule
+after the document title and PEP headers.
+Secondly, a callback transform recursively walks the document to create the TOC,
+starting from after the placeholder node.
+Whilst walking the document all reference nodes in the titles are removed, and
+titles are given a self-link.
+
+
+3.4.5 ``PEPFooter`` transform
+*****************************
+
+This firstly builds a map of file modification times from a single git call, as
+a speed-up. This will return incorrect results on a shallow check-out of the
+repository, as is the default on continuous integration systems.
+
+We then attempt to remove any empty references sections, and append metadata in
+the footer (source link and last modified timestamp).
+
+
+3.5 Prepare for writing
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+``pep_html_builder.FileBuilder.prepare_writing`` initialises the bare miniumum
+of the Docutils writer and the settings for writing documents.
+This provides a significant speed-up over the base Sphinx implementation as most
+of the data automatically initialised was unused.
+
+
+3.6 Translate Docutils to HTML
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``PEPTranslator`` overrides paragraph and reference logic to replicate
+processing from the previous ``docutils.writers.pep`` based system.
+Paragraphs are to be made compact where possible by omitting <p> tags, and
+footnote references should be enclosed in square brackets.
+
+
+3.7 Prepare for export to Jinja2
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Finally in ``pep_html_builder``, we gather all the parts to be passed to the
+Jinja2 template.
+This is also where we create the sidebar table of contents.
+
+The HTML files are then written out to the build directory.
+
+
+3. Theme
+--------
+
+The theme is comprised of the HTML template in
+``pep_sphinx_extensions/pep_theme/templates/page.html`` and the stylesheets in
+``pep_sphinx_extensions/pep_theme/static``.
+
+The template is entirely self-contained, not relying on any default behaviour
+from Sphinx.
+It specifies the CSS files to include, the favicon, and basic semantic
+information for document structure.
+
+The stylesheets are in two parts, with ``style.css`` defining the meat of the
+layout, and ``mq.css`` defining media-queries for a responsive layout.
```

Comment:
```suggestion
The stylesheet is split into two parts, with ``style.css`` defining the meat
of the layout, and ``mq.css`` adding media queries for a responsive design.
```

Fixed grammar/plural agreement issue (if you don't like that wording, I can offer others that also fix the problem). Also, corrected punctuation and multiple repetition issues.

Reply:
I've suggested an alternate, although it uses a two element list

A

Reply:
I see; I had one small suggested tweak on that but otherwise looks great :+1: 

(FYI, to make it easier on reviewers, you can stick your suggestions here too and even apply them, if you like—though on the other hand, that can be extra work for you)

Reply:
> (FYI, to make it easier on reviewers, you can stick your suggestions here too and even apply them, if you like—though on the other hand, that can be extra work for you)

Ahh, I was trying to avoid 5,000 commits by clicking the "commit suggestion" button for each change. Will keep in mind.